### PR TITLE
Update Build a Node Java guide

### PR DESCRIPTION
**Pull request by valentinewallace**
Our Java code samples were quite out of date. This updates them.

Next up: complete the "Using LDK" code samples and add a Rust-specific guide.

Supercedes #36

### CODE REVIEW

File: docs/build_node.md [hunk 1]
```diff
@@ -208,17 +221,19 @@ Filter tx_filter = Filter.new_impl(new Filter.FilterInterface() {
     }
 
     @Override
-    Option_C2Tuple_usizeTransactionZZ register_output(WatchedOutput output) {
+    public Option_C2Tuple_usizeTransactionZZ register_output(WatchedOutput output)
+    {
         // <insert code for you to watch for any transactions that spend this
-        // output on-chain>
-        // See blockdata.md for more info. XXX HOW2LINK???
```
Comment (TheBlueMatt) [inline docs/build_node.md]: This should ideally link to blockdata.md, I'm not sure how to insert a link.

File: docs/build_node.md [hunk 2]
```diff
@@ -305,22 +319,22 @@ byte[] best_block_hash = // <insert current chain tip block hash>;
 ChannelManagerConstructor channel_manager_constructor = new ChannelManagerConstructor(
   LDKNetwork.LDKNetwork_Bitcoin, UserConfig.default(), best_block_hash,
   block_height, keys_manager.as_KeysInterface(), fee_estimator, chain_monitor,
-  router, tx_broadcaster, logger);
+  tx_broadcaster, logger);
```
Comment (TheBlueMatt) [inline docs/build_node.md:322]: Huh?

File: docs/build_node.md (hunk 1 above)
Comment (valentinewallace) [reply docs/build_node.md]: I linked it below in References + Implementation Notes

File: docs/build_node.md [hunk 3]
```diff
@@ -503,9 +516,19 @@ The broader node-announcement message is not automatically broadcast.
...
 
 ```java
-// TODO(matt) :)
+byte[] rgb = // insert the RGB "color" of this node
+byte[] alias = // insert the alias of this node, i.e. its human readable name
+NetAddress[] addresses = // insert a list of socket addresses on which this node
+                         // accepts incoming connections
+
+// Once per minute or so, call:
+channel_manager.broadcast_node_announcement(rgb, alias, addresses);
```
Comment (TheBlueMatt) [inline docs/build_node.md:525]: This isn't really useful unless we can construct `NetAddress` objects, which we currently cannot.

File: docs/build_node.md (hunk 2 above)
Comment (valentinewallace) [reply docs/build_node.md:322]: https://github.com/lightningdevkit/ldk-garbagecollected/blob/main/src/main/java/org/ldk/batteries/ChannelManagerConstructor.java#L46 I don't see the router argument there anymore?

Comment (TheBlueMatt) [reply docs/build_node.md:322]: Yea, it'll be added soon :)

File: docs/build_node.md (hunk 3 above)
Comment (valentinewallace) [reply docs/build_node.md:525]: Ah okay. I'll remove the step for now, then?

File: docs/build_node.md (hunk 2 above)
Comment (valentinewallace) [reply docs/build_node.md:322]: Ah! Is it going to be optional? Otherwise, what should people put there if they skipped step 4?

Comment (TheBlueMatt) [reply docs/build_node.md:322]: Its `@Nullable`, yes.

File: docs/build_node.md (hunk 3 above)
Comment (TheBlueMatt) [reply docs/build_node.md:525]: Yea, feel free to drop it, I'll fix it soon, though, hopefully.

Comment (valentinewallace) [reply docs/build_node.md]: Updated!

**Comment by jkczyz** [comment]
> Next up: complete the "Using LDK" code samples and add a Rust-specific guide.

@valentinewallace I have a work in progress on the Rust part of this. Was a little distracted by the conference but should be able to finish up soon. We can coordinate offline.

**Comment by valentinewallace** [comment]
> > Next up: complete the "Using LDK" code samples and add a Rust-specific guide.
> 
> @valentinewallace I have a work in progress on the Rust part of this. Was a little distracted by the conference but should be able to finish up soon. We can coordinate offline.

@jkczyz Sounds good! Matt updated some of the "Using LDK" Java code samples in this PR if you wouldn't mind including those :) and he has more updates on the way

**Pull request merged**